contract/migrations/18.0.2.0.0/pre-migrate.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2025 ACSONE SA/NV
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+from openupgradelib import openupgrade
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    env.cr.execute(
+        """
+    DELETE FROM ir_model
+    WHERE model IN
+        ('contract.abstract.contract',
+        'contract.abstract.contract.line',
+        'contract.recurrency.mixin',
+        'contract.recurrency.basic.mixin'
+    );
+    """
+    )
```

Comment:
```suggestion
    )
    views_to_delete = ["contract.contract_template_line_form_view"]
    for view_ref in views_to_delete:
        view = env.ref(view_ref, raise_if_not_found=False)
        if view:
            view.unlink()
```

I got an error during the upgrade that **is_auto_renew** is not part of the module. This occures duing the update of **contract_line_form_view** which references **contract_abstract_contract_line_form_view**.

Reply:
The problem with this migration script is that you cannot delete this view if other views inherit from it.

Reply:
But you shouldn't remove views in the middle of the migration. OpenUpgrade already avoids a hard crash on that case, and they should be removed at the end of the cycle being noupdate=0. If they are not remvoed, it's because there are manual children views, and in that case, you should fix it manually on your DB.

Reply:
Is there a possibility to force the update of **contract_template_line_form_view** before **contract_line_form_view**  is updated? This could also fix the issue without removing/readding the contract_template_line_form_view.

Reply:
@pedrobaeza I got the same issue when launching a `click-odoo-update --update-all` on a DB that was migrated by Odoo from v12 to v18.
@CRogos I personally solved my issue by deleting all views (all entries in ir_ui_view table) before launching this big click-odoo-update, as in our case the users are not supposed to modify the views from the UI.

Reply:
FYI [delete_records_safely_by_xml_id](https://github.com/OCA/openupgradelib/blob/master/openupgradelib/openupgrade.py#L3428 ) contains `delete_childs` parameter to delete all inheriting views of the view you want to delete.

Reply:
@MiquelRForgeFlow solution is the best in my opinion, because everything which refers to this view should be recreated later anyway. (Anyone who is allowing to modify views in the UI can't be helped anyway 🤪 )